File: tests/mozmill/tests/testLinks/testHTMLAnchorElement/testLinkClick.js
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

var rpRootDir = "../../../";
var rpConst = require(rpRootDir + "lib/constants");
var rootDir = rpRootDir + rpConst.mozmillTestsRootDir;

var {assert, expect} = require(rootDir + "lib/assertions");
var prefs = require(rootDir + "lib/prefs");
var tabs = require(rootDir + "firefox/lib/tabs");

var rpUtils = require(rpRootDir + "lib/rp-utils");

const TEST_URL = "http://www.maindomain.test/link_1.html";


var setupModule = function(aModule) {
  aModule.controller = mozmill.getBrowserController();
  aModule.tabBrowser = new tabs.tabBrowser(aModule.controller);
  aModule.tabBrowser.closeAllTabs();

  prefs.setPref(rpConst.PREF_DEFAULT_ALLOW, false);
}

var teardownModule = function(aModule) {
  prefs.clearUserPref(rpConst.PREF_DEFAULT_ALLOW);
  aModule.tabBrowser.closeAllTabs();
}


var testLinkClick = function() {
  controller.open(TEST_URL);
  controller.waitForPageLoad();

  let link = rpUtils.getLink(controller);
  let linkURL = link.getNode().href;

  link.click();

  rpUtils.waitForTabLoad(controller, tabBrowser.getTab(0));

  var panel = tabBrowser.getTabPanelElement(0,
      '/{"value":"' + rpConst.REDIRECT_NOTIFICATION_VALUE + '"}');
  assert.ok(false === panel.exists(),
      "Following the link didn't cause a redirect");

  assert.equal(controller.tabs.activeTab.location.href, linkURL,
      "The location is correct.");
}

File: tests/marionette/tests/links/html_anchor_element/test_link_click.py
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.

from rp_ui_harness import RequestPolicyTestCase
from rp_puppeteer.ui.redirect_notification import RedirectNotification


TEST_URL = "http://www.maindomain.test/link_1.html";
PREF_DEFAULT_ALLOW = "extensions.requestpolicy.defaultPolicy.allow"


class TestLinkClick(RequestPolicyTestCase):
    def setUp(self):
        RequestPolicyTestCase.setUp(self)
        self.prefs.set_pref(PREF_DEFAULT_ALLOW, False);
        self.redir = RedirectNotification(lambda: self.marionette)


    def test_link_click(self):
        with self.marionette.using_context("content"):
            # load the test url
            self.marionette.navigate(TEST_URL)
            # get the link, its url, and click the link
            link = self.marionette.find_element("tag name", "a")
            link_url = link.get_attribute("href")
            link.click()

        self.assertFalse(self.redir.panel_exists(),
                         "Following the link didn't cause a redirect.")
        self.assertEqual(self.browser.tabbar.selected_tab.location,
                         link_url, "The location is correct.")
